src/Domain/LeanCode.DomainModels/Ids/TypedIdConverter.cs

```diff
@@ -23,6 +32,18 @@ public class IntTypedIdConverter<TId> : JsonConverter<TId>
 
     public override void Write(Utf8JsonWriter writer, TId value, JsonSerializerOptions options) =>
         writer.WriteNumberValue(value.Value);
+
+    public override TId ReadAsPropertyName(
+        ref Utf8JsonReader reader,
+        Type typeToConvert,
+        JsonSerializerOptions options
+    ) =>
+        reader.GetString() is string s && int.TryParse(s, out var id)
+            ? TId.Parse(id)
+            : throw new JsonException($"Could not deserialize {typeToConvert.Name}");
```

Comment:
Read to span, parse from span. Also let's use `{ return …; }` blocks for multiline expressions.